packages/mui-material/src/SpeedDial/SpeedDial.js

```diff
@@ -368,17 +376,24 @@ const SpeedDial = React.forwardRef(function SpeedDial(inProps, ref) {
 
   const children = allItems.map((child, index) => {
     const {
-      FabProps: { ref: origButtonRef, ...ChildFabProps } = {},
+      FabProps: { ref: origButtonRef } = {},
+      slotProps: childSlotProps = {},
       tooltipPlacement: tooltipPlacementProp,
     } = child.props;
 
+    const { fab: { ref: fabSlotOrigButtonRef, ...fabSlotProps } = {}, ...restOfSlotProps } =
+      childSlotProps;
+
     const tooltipPlacement =
       tooltipPlacementProp || (getOrientation(direction) === 'vertical' ? 'left' : 'top');
 
     return React.cloneElement(child, {
-      FabProps: {
-        ...ChildFabProps,
-        ref: createHandleSpeedDialActionButtonRef(index, origButtonRef),
+      slotProps: {
+        fab: {
+          ...fabSlotProps,
```

Comment:
Don't we need to destructure `ChildFabProps` here to keep `FabProps` working correctly?


```suggestion
        fab: {
          ...ChildFabProps,
          ...fabSlotProps,
```


I wonder why did test not break without it 🤔